File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int[] fib = new int[n + 1];
        fib[1] = 1;
        
        for (int i = 2; i <= n; i++)
            fib[i] = (fib[i-1] + fib[i-2])% 1234567;
        
        return fib[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    MOD = 1234567
    memo = [0] * (n + 1)
    
    for i in range(len(memo)):
        if i == 0 or i == 1:
            memo[i] = i
        else :
            memo[i] = (memo[i - 1] + memo[i - 2]) % MOD
            
    return memo[n]
